test/optimizer/plugins/ResolveValueSetComponentRuleURLsOptimizer.test.ts

```diff
@@ -0,0 +1,207 @@
+      const valueset = new ExportableValueSet('MyValueSet');
+      const rule = new ExportableValueSetFilterComponentRule(true);
+      rule.from = { system: 'http://hl7.org/fhir/observation-status' };
+      valueset.rules.push(rule);
+      const myPackage = new Package();
+      myPackage.add(valueset);
+
+      // Use a modified lake and fisher to force the local CS to have the same name
+      const modLake = cloneDeep(lake);
+      modLake.docs[0].content.name = 'ObservationStatus';
+      optimizer.optimize(myPackage, new MasterFisher(modLake, defs));
+
+      const expectedRule = new ExportableValueSetFilterComponentRule(true);
+      expectedRule.from = { system: 'http://hl7.org/fhir/observation-status' };
+      expect(valueset.rules).toContainEqual(expectedRule);
+    });
+
+    it('should replace filter rule valueset url with the name of a local ValueSet', () => {
+      const valueset = new ExportableValueSet('MyValueSet');
+      const rule = new ExportableValueSetFilterComponentRule(true);
+      rule.from = { valueSets: ['http://example.org/tests/ValueSet/simple.valueset'] };
+      valueset.rules.push(rule);
+      const myPackage = new Package();
+      myPackage.add(valueset);
+      optimizer.optimize(myPackage, fisher);
+
+      const expectedRule = new ExportableValueSetFilterComponentRule(true);
+      expectedRule.from = { valueSets: ['SimpleValueSet'] };
+      expect(valueset.rules).toContainEqual(expectedRule);
+    });
+
+    it('should replace concept rule valueset url with the name of a local ValueSet', () => {
+      const valueset = new ExportableValueSet('MyValueSet');
+      const rule = new ExportableValueSetConceptComponentRule(true);
+      rule.from = { valueSets: ['http://example.org/tests/ValueSet/simple.valueset'] };
+      valueset.rules.push(rule);
+      const myPackage = new Package();
+      myPackage.add(valueset);
+      optimizer.optimize(myPackage, fisher);
+
+      const expectedRule = new ExportableValueSetConceptComponentRule(true);
+      expectedRule.from = { valueSets: ['SimpleValueSet'] };
+      expect(valueset.rules).toContainEqual(expectedRule);
+    });
+
+    it('should replace filter rule valueset url with the name of a core FHIR CodeSystem', () => {
+      const valueset = new ExportableValueSet('MyValueSet');
+      const rule = new ExportableValueSetFilterComponentRule(true);
+      rule.from = { valueSets: ['http://hl7.org/fhir/ValueSet/observation-status'] };
+      valueset.rules.push(rule);
+      const myPackage = new Package();
+      myPackage.add(valueset);
+      optimizer.optimize(myPackage, fisher);
+
+      const expectedRule = new ExportableValueSetFilterComponentRule(true);
+      expectedRule.from = { valueSets: ['ObservationStatus'] };
+      expect(valueset.rules).toContainEqual(expectedRule);
+    });
+
+    it('should not replace filter rule system url with the name of a core FHIR ValueSet when it is same as local code system name', () => {
```

Comment:
It took me a second to figure out why this is necessary, but you're right: SUSHI wouldn't be happy being told `from valueset ABC` when `ABC` is the name of a local CodeSystem.